src/main/java/com/ning/http/client/HttpResponseBodyPart.java

```diff
@@ -86,4 +88,13 @@ public boolean isLast() {
      * @return {@link ByteBuffer}
      */
     public abstract ByteBuffer getBodyByteBuffer();
+
+    /**
+     * Retrieves a {@link PauseHandler} to pause/resume event processing. Useful when an {@link AsyncHandler}'s resources are
+     * saturated, and it can't handle the body part temporarily. If the handler uses the {@link PauseHandler#requestPause()}
+     * method, it's also responsible for calling {@link PauseHandler#resume()} when it could handle the part correctly.
+     *
+     * @return a {@link PauseHandler} to pause the events processing in the current http transaction.
+     */
+    public abstract PauseHandler getPauseHandler();
```

Comment:
Do you think it could be possible for someone other than Grizzly to be extending this abstract class? should we give it a default implementation just in case?

Reply:
I don't think so, but it's possible. I added a default implementation that throws an exception.
In the same line, I added back the constructor of GrizzlyResponseBodyPart that receives a Connection, because it's POSSIBLE that somebody uses it (again, I don't think so)